### PR TITLE
BAU Refactor to use the new wizard

### PR DESCRIPTION
**Pull request by MattBidewell**
### Issue tracking

- [KBV-XXXX](https://govukverify.atlassian.net/browse/KBV-XXXX)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

### CODE REVIEW

File: src/app/address/controllers/address.js [hunk 1]
```diff
@@ -63,39 +49,16 @@ class AddressController extends BaseController {
 
   async saveValues(req, res, callback) {
     super.saveValues(req, res, () => {
-      const isPreviousAddress = req.originalUrl.startsWith("/previous/address");
```
Comment (MattBidewell) [inline src/app/address/controllers/address.js:66]: ripping out any refence to /previous

File: src/app/address/controllers/address.test.js [hunk 2]
```diff
@@ -42,7 +42,7 @@ describe("address controller", () => {
       const generatedAddress = addressFactory(1);
       req.sessionModel.set("addressPostcode", generatedAddress[0].postalCode);
 
-      req.sessionModel.set("chosenAddress", generatedAddress[0]);
```
Comment (MattBidewell) [inline src/app/address/controllers/address.test.js:45]: standardised the naming, some places we where using chosenAddress and some places, address.

Stuck with just address. (dont ask me why I used chosenAddress in the first place!)

File: src/app/address/controllers/address.test.js [hunk 3]
```diff
@@ -58,45 +58,6 @@ describe("address controller", () => {
...
-          addressHouseName: generatedAddress[1].buildingName,
-          addressStreetName: generatedAddress[1].streetName,
-          addressLocality: generatedAddress[1].addressLocality,
-          addressYearFrom: Number(generatedAddress[1].validFrom),
-          addressFlatNumber: generatedAddress[1].buildingNumber,
-        })
-      );
-    });
-
-    it("should prepopulate with the current address when in editing current address route", () => {
```
Comment (MattBidewell) [inline src/app/address/controllers/address.test.js:82]: Loads of tests are now no longer needed. As we dont need to tests specific logic when /previous/edit is hit.

File: src/app/address/controllers/addressConfirm.js [hunk 4]
```diff
@@ -15,47 +15,44 @@ const {
 class AddressConfirmController extends BaseController {
   locals(req, res, callback) {
     super.locals(req, res, (err, locals) => {
-      const sessionAddresses = req.sessionModel.get("addresses");
+      const currentAddress = req.sessionModel.get("address");
+      const previousAddress = req.session["hmpo-wizard-previous"]?.address;
```
Comment (MattBidewell) [inline src/app/address/controllers/addressConfirm.js:19]: Accessing the alternative wizard like this for now.

In the future we could make the confirmation step a third wizard that orchestrates the others?
As in the future we may wish to get more than one previous address. Food for thought.

File: src/app/address/controllers/address.js [hunk 5]
```diff
@@ -109,12 +72,12 @@ class AddressController extends BaseController {
       addressLocality,
       addressYearFrom,
     } = undefined,
```
Comment (jkunle) [inline src/app/address/controllers/address.js:74]: I thought it is not good practice to assign `undefined` I would think null would be appropriate

File: src/app/address/controllers/addressConfirm.js (hunk 4 above)
Comment (merlinc) [reply src/app/address/controllers/addressConfirm.js:19]: Yeah, that would be my thought - two wizards `current` and `previous` using mostly the same config, feeding into a `summary` one. 

This is how the summary screen on the HMPO journey works, feeding back into other wizards for editing as needed, invalidating fields if values changes (e.g. if year changes to 2000, then the previous address data is now invalid).

**Review by merlinc — APPROVED** [review]
Lots of simplifying refactoring going on here :+1:

**Pull request merged**